babel-plugin-optimize-obj-str/readme.md

```diff
@@ -0,0 +1,222 @@
+# babel-plugin-optimize-obj-str
+
+> [Babel](https://babeljs.io/) plugin to optimize [`obj-str`](../) calls by replacing them with an equivalent unrolled expression.
+
+Even though the `obj-str` function is negligible in size over-the-wire, the motivation for this plugin is that transformed expressions [execute almost twice as fast as equivalent calls.](https://jsbench.me/nukl0mvqze/1)
+
+```js
+import objstr from 'obj-str';
+objstr({
+  'my-classname': true,
+  'another-one': maybe,
+});
+
+// Transformed:
+'' + (true ? 'my-classname' : '') + (maybe ? ' another-one' : '');
+```
+
+> **Note**: This plugin does not attempt to minimize output, so it may be wordy. You should additionally use a minifier like [`terser`](https://terser.org/) for best results.
+
+## Install
+
+```
+npm install --save-dev babel-plugin-optimize-obj-str
+```
+
+## Usage
+
+**Via babel.config.js** ([recommended](https://babeljs.io/docs/en/configuration)):
+
+```js
+module.exports = {
+  plugins: ['optimize-obj-str'],
+};
+```
+
+**Via CLI**:
+
+```
+babel --plugins optimize-obj-str script.js
+```
+
+## Options
+
+```js
+module.exports = {
+  plugins: [
+    [
+      'optimize-obj-str',
+      {
+        strict: false,
+        leadingSpace: false,
+        referencesImport: ['obj-str', 'default'],
+      },
```

Comment:
On `amphtml`'s end we'll configure as:

```js
{
  strict: true,
  leadingSpace: true,
}
```

I added these options since `false` seems like a more reasonable default, but that could be argued.

Alternatively, we could remove any of these options and simplify, but we'd ultimately the behavior to match like `strict: true` and `leadingSpace: true`.

Reply:
Given that user-agents don't have a strong preference around `leadingSpace` and the output is significantly more terse when it's set to true... I'd lean on `true` being the default.

What's your thoughts @lukeed?

Reply:
I'm leaning on forcing it to be `true` & removing the option entirely. The output without it makes it easy to outweigh the size of `obj-str`